lib/slave.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SLAVE_HPP
+#define SLAVE_HPP
+
+#include <array>
+
+#include "wrap-hwlib.hpp"
+
+#include "address.hpp"
+#include "frame.hpp"
+#include "parity.hpp"
+#include "uart_lib.hpp"
+
+namespace LinkModule {
+class Slave {
+    UARTLib::UARTConnection &uart;
+
+  public:
+    /**
+     * @brief Construct a new Slave object
+     *
+     * @param uart UARTConnection to communicate with
+     */
+    Slave(UARTLib::UARTConnection &uart);
+
+    /**
+     * @brief Blocks operations until an address is received
+     *
+     * @details
+     * Blocks operation until an address is received.
+     * When timed out, a bool is returned signaling
+     * the timeout.
+     *
+     * @param timeoutUs
```

Comment:
Parameter is unexplained but definition can be derived in a straight-forward manner. Change to your own liking.